### PR TITLE
mimxrt: Add the Adafruit Metro M7 board.

### PR DESCRIPTION
**Pull request by robert-hh**
Support for WiFi and BLE is prepared and will be provided with separate pull requests.

### CODE REVIEW

File: ports/mimxrt/boards/ADAFRUIT_METRO_M7/mpconfigboard.h [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+// Wifi Deinitions
+#define MICROPY_PY_NETWORK_HOSTNAME_DEFAULT "mpy-metro-m7"
+
+#define MICROPY_HW_WIFI_SPI_ID       (0)
+#define MICROPY_HW_WIFI_SPI_BAUDRATE (8000000)
+
+#define MICROPY_HW_NINA_ACK          pin_find(mp_obj_new_str("ESP_BUSY", 8))
+#define MICROPY_HW_NINA_CS           pin_find(mp_obj_new_str("ESP_CS", 6))
+#define MICROPY_HW_NINA_RESET        pin_find(mp_obj_new_str("ESP_RESET", 9))
```
Comment (dpgeorge) [inline ports/mimxrt/boards/ADAFRUIT_METRO_M7/mpconfigboard.h:56]: I'm not sure this is the right way to get a pin, it's quite inefficient.  Why not just use `&pin_GPIO_AD_07`?

Comment (robert-hh) [reply ports/mimxrt/boards/ADAFRUIT_METRO_M7/mpconfigboard.h:56]: I've had chosen that way for better readability in the file. This part of the code is only executed once while the interface is set-up, so run-time efficiency does not matter. It requires a little bit more code, which is not a problem at the M7 board. But I can of course change it.

Comment (dpgeorge) [reply ports/mimxrt/boards/ADAFRUIT_METRO_M7/mpconfigboard.h:56]: Would this work: `pin_find(MP_QSTR_ESP_BUSY)`?  At least then it doesn't need to allocate the string.

Or maybe mimxrt needs named pins like stm32, then you could do `pyb_pin_ESP_BUSY` (probably a better name though, not pyb).

Comment (dpgeorge) [reply ports/mimxrt/boards/ADAFRUIT_METRO_M7/mpconfigboard.h:56]: Actually it should be `pin_find(MP_OBJ_NEW_QSTR(MP_QSTR_ESP_BUSY))`.

Comment (robert-hh) [reply ports/mimxrt/boards/ADAFRUIT_METRO_M7/mpconfigboard.h:56]: The latter works, and WiFi starts with it.

> Or maybe mimxrt needs named pins like stm32, then you could do pyb_pin_ESP_BUSY (probably a better name though, not pyb).

At the moment the pin names as like `pin_GPIOxxxxx`. But reworking the Pin section is another huge task on the t.b.d. list.

Comment (dpgeorge) [reply ports/mimxrt/boards/ADAFRUIT_METRO_M7/mpconfigboard.h]: OK, that looks good then.  Happy to merge this now.

> But reworking the Pin section is another huge task on the t.b.d. list.

I'd suggest holding back on that.  I'd rather consolidate all ports to use the same pins.csv mechanism.

Comment (robert-hh) [reply ports/mimxrt/boards/ADAFRUIT_METRO_M7/mpconfigboard.h]: > I'd suggest holding back on that. I'd rather consolidate all ports to use the same pins.csv mechanism.

That looks huge.

Comment (dpgeorge) [reply ports/mimxrt/boards/ADAFRUIT_METRO_M7/mpconfigboard.h]: I'm not suggesting that you do it.  It was on @jimmo 's list for a while now.

**Pull request merged**